### PR TITLE
Raspberry Pi: Add SNMP extend to monitor IO pins or sensor modules connected to the GPIO header

### PR DESCRIPTION
**Pull request by nightcore500**
This patch adds a SNMP extend script to monitor IO pins or sensor modules connected to the GPIO header.
The respective sensors can be configured individually with a configuration file.
The reading of IO contacts is possible by default if wiringpi is installed on the respective device. 
For sensor modules an additional program or a script is necessary which can be specified in the configuration. 
This allows to monitor practically everything connected to the GPIO header.
The configuration can also be checked using "rpigpiomonitor.php -validate".

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by nightcore500** [comment]
This PR belongs to https://github.com/librenms/librenms-agent/pull/364 in the librenms-agent repo.

**Comment by nightcore500** [comment]
Ready for review :)

File: includes/discovery/sensors/rpigpiomonitor.inc.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        }
+    } else {
+        $sensors[$sensor_index]['value'] = intval($entry['nsExtendOutLine']);
+        $sensors[$sensor_index]['oid'] = '.1.3.6.1.4.1.8072.1.3.2.4.1.2.' . $index;
+        $sensor_index++;
+    }
+}
+
+echo "[rpigpiomonitor] Sensor Array dump: \n";
+var_dump($sensors);
```
Comment (murrant) [inline includes/discovery/sensors/rpigpiomonitor.inc.php:69]: debug output, please use d_echo($sensors) or remove.

Comment (nightcore500) [reply includes/discovery/sensors/rpigpiomonitor.inc.php]: Done :)

**Comment by nightcore500** [comment]
@murrant 
Do you have time for another review? :)

**Comment by nightcore500** [comment]
@murrant @Jellyfrog 
I would like to ask again for a review. Would either of you have time?

**Comment by Jellyfrog** [comment]
(Rerunning tests)

**Comment by nightcore500** [comment]
@Jellyfrog Do you have an idea why the IP test fails?

**Comment by Jellyfrog** [comment]
Yes, https://github.com/librenms/librenms/pull/13468

**Comment by nightcore500** [comment]
> @Jellyfrog Do you have an idea why the IP test fails?



> Yes, #13468

Ah, thanks. I just noticed the PR after submitting ^^

**Review by murrant — APPROVED** [review]

**Pull request merged**